packages/gensx/src/dev-server.ts

```diff
@@ -308,296 +274,278 @@ export class GensxServer {
-          createdAt: execution.createdAt,
-        };
+    this.app.get(`/workflows/:workflowName/executions/:executionId`, (c) => {
+      const workflowName = c.req.param("workflowName");
+      const executionId = c.req.param("executionId");
+
+      // Will throw NotFoundError if execution doesn't exist or doesn't match workflow
+      const execution = this.getExecutionOrThrow(executionId, workflowName);
+
+      // Construct the response data with proper type safety
+      const responseData: {
+        id: string;
+        executionStatus: ExecutionStatus;
+        createdAt: string;
+        finishedAt?: string;
+        output?: unknown;
+        error?: string;
+      } = {
+        id: execution.id,
+        executionStatus: execution.executionStatus,
+        createdAt: execution.createdAt,
+      };
 
-        // Add optional fields if they exist
-        if (execution.finishedAt) {
-          responseData.finishedAt = execution.finishedAt;
-        }
+      // Add optional fields if they exist
+      if (execution.finishedAt) {
+        responseData.finishedAt = execution.finishedAt;
+      }
 
-        if (execution.output !== undefined) {
-          responseData.output = execution.output;
-        }
+      if (execution.output !== undefined) {
+        responseData.output = execution.output;
+      }
 
-        if (execution.error) {
-          responseData.error = execution.error;
-        }
+      if (execution.error) {
+        responseData.error = execution.error;
+      }
 
-        return c.json({
-          status: "ok",
-          data: responseData,
-        });
-      },
-    );
+      return c.json({
+        status: "ok",
+        data: responseData,
+      });
+    });
 
     // List executions for a workflow
-    this.app.get(
-      `/org/${this.org}/projects/${this.project}/workflows/:workflowName/executions`,
-      (c) => {
-        const workflowName = c.req.param("workflowName");
-        // Will throw NotFoundError if workflow doesn't exist
-        this.getWorkflowOrThrow(workflowName);
-
-        // Filter executions for this workflow
-        const executions = Array.from(this.executionsMap.values())
-          .filter((exec) => exec.workflowName === workflowName)
-          // Sort by createdAt in descending order (newest first)
-          .sort(
-            (a, b) =>
-              new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
-          )
-          .map((exec) => ({
-            id: exec.id,
-            executionStatus: exec.executionStatus,
-            createdAt: exec.createdAt,
-            finishedAt: exec.finishedAt,
-          }));
+    this.app.get(`/workflows/:workflowName/executions`, (c) => {
+      const workflowName = c.req.param("workflowName");
+      // Will throw NotFoundError if workflow doesn't exist
+      this.getWorkflowOrThrow(workflowName);
+
+      // Filter executions for this workflow
+      const executions = Array.from(this.executionsMap.values())
```

Comment:
Should this hit our API if there is an API key?

Reply:
I'm not sure i'm following. Why would the dev server hit our API?

Reply:
Sorry I was confused. This is executions not traces.